internal/nodes/ingress_model_rel.go

```diff
@@ -235,35 +242,37 @@ func GatewayChanges(gwName string, namespace string, key string) ([]string, bool
 	return allGateways, true
 }
 
-func GatewayClassChanges(gwClassName string, namespace string, key string) ([]string, bool) {
+func GWClassToGateway(gwClassName string, namespace string, key string) ([]string, bool) {
 	found, gateways := objects.ServiceGWLister().GetGWclassToGateways(gwClassName)
 	return gateways, found
 }
 
 func IngressChanges(ingName string, namespace string, key string) ([]string, bool) {
 	var ingresses []string
 	ingresses = append(ingresses, ingName)
-	myIng, err := utils.GetInformers().IngressInformer.Lister().ByNamespace(namespace).Get(ingName)
+	ingObj, err := utils.GetInformers().IngressInformer.Lister().Ingresses(namespace).Get(ingName)
 
 	if err != nil {
 		// Detect a delete condition here.
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			// Remove all the Ingress to Services mapping.
 			// Remove the references of this ingress from the Services
 			objects.SharedSvcLister().IngressMappings(namespace).RemoveIngressMappings(ingName)
+			objects.SharedSvcLister().IngressMappings("").RemoveIngressClassMappings(ingName)
 		}
 	} else {
-		ingObj, ok := utils.ToNetworkingIngress(myIng)
-		if !ok {
-			utils.AviLog.Errorf("Unable to convert obj type interface to networking/v1beta1 ingress")
-		}
-
 		// simple validator check for duplicate hostpaths, logs Warning if duplicates found
 		success := validateSpecFromHostnameCache(key, ingObj.Namespace, ingObj.Name, ingObj.Spec)
 		if !success {
 			return ingresses, false
 		}
 
+		if ingObj.Spec.IngressClassName != nil {
+			objects.SharedSvcLister().IngressMappings("").UpdateIngressClassMappings(namespace+"/"+ingName, *ingObj.Spec.IngressClassName)
```

Comment:
Since ingressClass is not namespace scoped so the Ingressclass mappings in this SharedSvcLister reside in an empty string key. so the cache looks something like this  for ingressClass->ing mapping
`cache: {"": {"avigwclass1":  ["ns1/ing1", "ns2/ing2"...]}}`